lua/wire/flir.lua

```diff
@@ -37,126 +56,158 @@ if CLIENT then
 		["$pp_colour_brightness"] = 0
 	}
 
-	local function SetFLIRMat(ent)
+
+
+	--add and remove entities from the FLIR rendering stack
+	local function SetFLIR(ent)
 		if not IsValid(ent) then return end
+		local c = ent:GetClass()
 
-		if (ent:GetMoveType() == MOVETYPE_VPHYSICS or ent:IsPlayer() or ent:IsNPC() or ent:IsRagdoll() or ent:GetClass() == "gmod_wire_hologram") and ent:GetColor().a > 0 then
-			ent.RenderOverride = FLIR.Render
-			FLIR.RenderStack[ent] = true
+		if (c == "prop_physics" or ent:GetMoveType() == MOVETYPE_VPHYSICS or (ent:IsPlayer() and ent != ply) or ent:IsNPC() or ent:IsRagdoll() or c == "gmod_wire_hologram") and ent:GetColor().a > 0 then
+			if ent:GetColor().a > 0 then
+				FLIR.RenderStack[ent] = true
+				ent.RenderOverride = FLIR.Render	--we're already rendering later, so don't bother beforehand
+				ent:CallOnRemove("RemoveFLIR", RemoveFLIR)
```

Comment:
RemoveFLIR is nil here since it is a local and below where this function is. Also, CallOnRemove in clientside will trigger sometimes even if the entity isn't actually removed. Will need to check if the entity is valid in a timer to make sure it was actually removed or not.